### PR TITLE
only-show-the-red-underline-if-both-yellow-and-red-are-present

### PR DESCRIPTION
**Pull request by farfaraji**

### CODE REVIEW

**Review by aterrel — CHANGES_REQUESTED** [review]
Can you add an accompanying test to the fix?

File: pylsp/python_lsp.py [hunk 1]
```diff
@@ -298,11 +298,23 @@ def hover(self, doc_uri, position):
...
+        errors = []
+        new_lints = []
+        for lint in lints:
+            new_lints.append(lint)
+            if lint['severity'] == lsp.DiagnosticSeverity.Error:
+                errors.append(lint)
+            else:
+                for error in errors:
+                    if lint['range']['start']['line'] == error['range']['start']['line'] or \
+                       lint['range']['start']['line'] == error['range']['end']['line']:
```
Comment (aterrel) [inline pylsp/python_lsp.py]: What if you have an error between the lint? Shouldn't this be:

```
lint_start, lint_end = lint['range']['start']['line'],  lint['range']['end']['line']
error_start, error_end = error['range']['start']['line'],  error['range']['end']['line'] 
if error_start <= lint_end <= error_end or \
    error_start <= lint_start <= error_end:
```

also seems like you have a race condition, what if the lint comes before the error? perhaps you want :

```
errors = [lint in lints if lint['severity'] == lsp.DiagnosticSeverity.Error]
new_lints = []
for lint in lints:
    for  error in errors:
        lint_start, lint_end = lint['range']['start']['line'],  lint['range']['end']['line']
        error_start, error_end = error['range']['start']['line'],  error['range']['end']['line'] 
        if not (error_start <= lint_end <= error_end or error_start <= lint_start <= error_end):    
            new_lints.append(lint)
```

File: pylsp/python_lsp.py [hunk 2]
```diff
@@ -291,18 +291,28 @@ def hover(self, doc_uri, position):
...
+        lints = flatten(self._hook('pylsp_lint', doc_uri, is_saved=is_saved))
+        # korbit_custom_lints(lints)
+        # errors = [lint for lint in lints if lint['severity'] == lsp.DiagnosticSeverity.Error]
+        # new_lints = []
+        # for lint in lints:
+        #     for error in errors:
+        #         lint_start, lint_end = lint['range']['start']['line'], lint['range']['end']['line']
+        #         error_start, error_end = error['range']['start']['line'], error['range']['end']['line']
+        #         if not (error_start <= lint_end <= error_end or error_start <= lint_start <= error_end):
+        #             new_lints.append(lint)
```
Comment (antoineKorbit) [inline pylsp/python_lsp.py]: Delete ?

File: pylsp/python_lsp.py [hunk 3]
```diff
@@ -475,3 +476,15 @@ def flatten(list_of_lists):
 
 def merge(list_of_dicts):
     return {k: v for dictionary in list_of_dicts for k, v in dictionary.items()}
+
+
+def korbit_custom_lints(lints):
+    errors = [lint for lint in lints if lint['severity'] == lsp.DiagnosticSeverity.Error]
```
Comment (antoineKorbit) [inline pylsp/python_lsp.py:482]: So we filter out non `lsp.DiagnosticSeverity.Error`, what are the other types of possibilities that we want to ignore here?

Comment (antoineKorbit) [reply pylsp/python_lsp.py:482]: Does it mean that we aren't showing yellow highlight ? Or is it just for the one that are on the same line ?

Comment (antoineKorbit) [reply pylsp/python_lsp.py:482]: Should you add a test to make sure that on 2 different lines, we are still showing the `lsp.DiagnosticSeverity.Warning` ?

**Review by antoineKorbit — CHANGES_REQUESTED** [review]

**Review by aterrel — APPROVED** [review]

**Review by antoineKorbit — APPROVED** [review]

**Pull request merged**